CMakeLists.txt

```diff
@@ -747,6 +747,10 @@ option(SWIFT_ENABLE_EXPERIMENTAL_PARSER_VALIDATION
   "Enable experimental SwiftParser validation by default"
   FALSE)
 
+option(SWIFT_ENABLE_EXPERIMENTAL_POINTER_BOUNDS
```

Comment:
What is the main motivation to make this a compile time flag rather than just a runtime switch for the compiler?

Reply:
Any reason we want to keep this out of stdlib by default? My main concern is that we are not really testing this in the CI if this is off by default (unless we make this on in the CI).

Reply:
I mostly followed what other experimental features did, so I'm not married to the approach. But I do have a concern that we'll want to make breaking changes to the macro API, so we need some way of making it clear that users shouldn't use this macro directly and expect source compatibility.
Note that this particular change is from https://github.com/swiftlang/swift/pull/76969, so we might want to take the discussion there.

Reply:
Ah, I see. So Swift has no mechanism equivalent to "unstable" features and APIs in Rust that can break any time? Now I better understand the motivation and if that is the case (no alternative), I am OK with this approach.

Reply:
I'm not sure tbh, I'm completely new to swift. @DougGregor could you fill us in?

Reply:
Right, so implicitly applying the macro on import could be an unstable language feature, but there's no way to include the macro in the stdlib in an "unstable" way, meaning that has to be stabilised before the importing?